### PR TITLE
Build docs separate

### PR DESCRIPTION
**Pull request by NathanDunfield**
Build the wheels with no docs and later use a separate script to build the docs.  For the actual release, another script would “manually” insert the docs into the various snappy wheels (which are just zip files) before uploading to PyPI.

### CODE REVIEW

File: doc_src/add_doc_to_wheel.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import shutil
+
+doc_zipfile = os.path.abspath(sys.argv[1])
+wheel_path = os.path.abspath(sys.argv[2])
+
+if not (doc_zipfile.endswith('.zip') and wheel_path.endswith('.whl')):
+    raise ValueError('Usage: add_doc_to_wheel docs.zip snappy.whl')
+
+python = sys.executable
+tmp_dir = tempfile.mkdtemp()
```
Comment (unhyperbolic) [inline doc_src/add_doc_to_wheel.py:21]: with tempfile.TemporaryDirectory() as tmp_dir:

might be more pythonic. Not sure whether cleaning up the tmp directory at the end is a feature (making debugging easier) or bug.

File: .github/workflows/sage_test.yml [hunk 2]
```diff
@@ -47,3 +47,15 @@ jobs:
...
+        # Doesn't just do "make html" as neither "make" nor "zip" is
+        # installed in this image.
+        run: |
+          cd /__w/SnapPy/SnapPy/doc_src
+          sage -python -m sphinx -b html -E -d _build/doctrees . _build/doc
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: doc
+          path: /__w/SnapPy/SnapPy/doc_src/_build/doc/*
```
Comment (unhyperbolic) [inline .github/workflows/sage_test.yml:61]: It might be a mess to download all that stuff. Would it be hard to install zip in the container temporarily?

**Review by unhyperbolic — APPROVED** [review]

File: doc_src/add_doc_to_wheel.py (hunk 1 above)
Comment (NathanDunfield) [reply doc_src/add_doc_to_wheel.py:21]: It was intended to make debugging easier.

File: .github/workflows/sage_test.yml (hunk 2 above)
Comment (NathanDunfield) [reply .github/workflows/sage_test.yml:61]: The `upload-artifact` action automatically puts the files in a single zipfile for easy download.

**Pull request merged**